pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -0,0 +1,204 @@
+
+	default:
+		klog.Errorf("[topologymanager] Unknow policy %s, using default policy %s", topologyPolicyName, PolicyPreferred)
+		policy = NewPreferredPolicy()
+	}
+
+	var hp []HintProvider
+	pnh := make(map[string]containers)
+	pm := make(map[string]string)
+	manager := &manager{
+		hintProviders:    hp,
+		podTopologyHints: pnh,
+		podMap:           pm,
+		policy:           policy,
+	}
+
+	return manager
+}
+
+func (m *manager) GetAffinity(podUID string, containerName string) TopologyHint {
+	return m.podTopologyHints[podUID][containerName]
+}
+
+func (m *manager) calculateTopologyAffinity(pod v1.Pod, container v1.Container) (TopologyHint, bool) {
+	admitPod := true
+	firstHintProvider := true
+	var containerHints []TopologyHint
+	for _, hp := range m.hintProviders {
+		topologyHints, admit := hp.GetTopologyHints(pod, container)
+		if !admit && topologyHints == nil {
+			klog.Infof("[topologymanager] Hint Provider does not care about this container")
+			continue
+		}
+		var tempMask []TopologyHint
+		for _, hint := range topologyHints {
+			klog.Infof("Hint: %v", hint)
+			if firstHintProvider {
+				tempMask = append(tempMask, hint)
+			} else {
+				for _, storedHint := range containerHints {
+					if storedHint.SocketMask.IsEqual(hint.SocketMask) {
+						klog.Infof("Masks equal")
+						tempMask = append(tempMask, hint)
+						break
+					}
+				}
+			}
+		}
+		containerHints = tempMask
+		tempMask = nil
+		firstHintProvider = false
+
+		klog.Infof("ContainerHints: %v", containerHints)
+
+	}
+	filledMask, _ := socketmask.NewSocketMask()
+	filledMask.Fill()
+	numBitsSet := filledMask.Count()
+	var containerTopologyHint TopologyHint
+	for _, hint := range containerHints {
+		if hint.SocketMask.Count() < numBitsSet {
+			numBitsSet = hint.SocketMask.Count()
+			containerTopologyHint = hint
+		}
+	}
+
+	if containerTopologyHint.SocketMask.Count() > 1 {
+		klog.Infof("Cross Socket Affinity.")
+		admitPod = false
+	}
+
+	klog.Infof("[topologymanager] ContainerTopologyHint: %v AdmitPod: %v", containerTopologyHint, admitPod)
+
+	return containerTopologyHint, admitPod
+}
+
+func (m *manager) AddHintProvider(h HintProvider) {
+	m.hintProviders = append(m.hintProviders, h)
+}
+
+func (m *manager) AddContainer(pod *v1.Pod, containerID string) error {
+	m.podMap[containerID] = string(pod.UID)
+	return nil
+}
+
+func (m *manager) RemoveContainer(containerID string) error {
+	podUIDString := m.podMap[containerID]
+	delete(m.podTopologyHints, podUIDString)
+	delete(m.podMap, containerID)
+	klog.Infof("[topologymanager] RemoveContainer - Container ID: %v podTopologyHints: %v", containerID, m.podTopologyHints)
+	return nil
+}
+
+func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	klog.Infof("[topologymanager] Topology Admit Handler")
+	pod := attrs.Pod
+	c := make(containers)
+	klog.Infof("[topologymanager] Pod QoS Level: %v", pod.Status.QOSClass)
+
+	qosClass := pod.Status.QOSClass
+
+	if qosClass == "Guaranteed" {
```

Comment:
Do we need the intermediate `qosClass` variable here?  Also, I probably wouldn't use the raw string "Guaranteed" when doing this comparison. The `qosClass` varaible is only used once and it's already pretty clear what it's meant for if used as `pod.Status.QOSClass == v1.PodQOSGuaranteed`.